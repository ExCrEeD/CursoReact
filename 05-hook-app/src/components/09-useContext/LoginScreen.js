import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext);
    const usr = {
        id: 123,
        name:'daniel'
    }
    return (
        <div>
            <h1>login screen</h1>
            <button
                className="btn btn-primary"
                onClick={()=> setUser(usr)}
            >
                Login
            </button>
        </div>
    )
}
