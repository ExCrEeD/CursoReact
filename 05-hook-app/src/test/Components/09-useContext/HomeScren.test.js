import { mount } from "enzyme";u
import { HomeScreen } from "../../../components/09-useContext/HomeScreen"
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('pruebas en <HomeScreen/>', () => {
    const user = {
        name:'daniel',
        email:'proof@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen/>
        </UserContext.Provider>);
    
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

})
