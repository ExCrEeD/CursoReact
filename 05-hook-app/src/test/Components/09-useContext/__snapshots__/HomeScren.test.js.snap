// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomeScreen/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home screen
  </h1>
  <pre>
    {
   "name": "daniel",
   "email": "proof@gmail.com"
}
  </pre>
</div>
`;
